src/error.rs

```diff
@@ -48,7 +48,7 @@ impl FromEnvError {
             FromEnvErrorInner::CannotOpenPath(..) => FromEnvErrorKind::CannotOpenPath,
             FromEnvErrorInner::CannotOpenFd(..) => FromEnvErrorKind::CannotOpenFd,
             FromEnvErrorInner::NegativeFd(..) => FromEnvErrorKind::NegativeFd,
-            FromEnvErrorInner::NotAPipe(..) => FromEnvErrorKind::NotAPipe,
+            FromEnvErrorInner::NotAPipeOrSocket(..) => FromEnvErrorKind::NotAPipeOrSocket,
```

Comment:
This would probably be an undesired breaking change, don't think we can rename this enum variant which is unfortunate

Could add a `NotASocket`